### PR TITLE
SAA-590: Calculate earliest release date

### PR DESCRIPTION
**Pull request by smcveigh941**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/util/EarliestReleaseDateHelper.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      hasTariffDate(prisoner) -> prisoner.tariffDate
+      else -> null
+    },
+    isTariffDate = prisoner.tariffDate?.isEqual(releaseDate) ?: false,
+    isConvictedUnsentenced = prisoner.legalStatus == Prisoner.LegalStatus.CONVICTED_UNSENTENCED,
+    isImmigrationDetainee = prisoner.legalStatus == Prisoner.LegalStatus.IMMIGRATION_DETAINEE,
+    isRemand = prisoner.legalStatus == Prisoner.LegalStatus.REMAND,
+    isIndeterminateSentence = prisoner.legalStatus == Prisoner.LegalStatus.INDETERMINATE_SENTENCE,
+  )
+}
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/util/EarliestReleaseDateHelper.kt:27]: Were the rules here provided from somewhere?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ActivityScheduleServiceTest.kt [hunk 2]
```diff
@@ -451,9 +564,19 @@ class ActivityScheduleServiceTest {
...
-    whenever(prisonApiClient.getPrisonerDetails("123456", fullInfo = false)).doReturn(Mono.just(prisoner))
-    whenever(prisonApiClient.getPrisonerDetails("123456", fullInfo = false)).doReturn(Mono.just(prisoner))
+    whenever(prisonPayBandRepository.findByPrisonCode(caseLoad)).thenReturn(
+      prisonPayBandsLowMediumHigh(caseLoad),
+    )
+    whenever(prisonApiClient.getPrisonerDetails("123456", fullInfo = false)).doReturn(
+      Mono.just(
+        prisoner,
+      ),
+    )
```
Comment (timharrison-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ActivityScheduleServiceTest.kt:574]: Very minor one. Just wondering on the format? Could be one line?
`whenever(prisonApiClient.getPrisonerDetails("123456", fullInfo = false)).doReturn(Mono.just(prisoner))`

**Review by timharrison-moj — APPROVED** [review]
LGTM. Very minor questions.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/util/EarliestReleaseDateHelper.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package uk.gov.justice.digital.hmpps.hmppsactivitiesmanagementapi.util
+
+import uk.gov.justice.digital.hmpps.hmppsactivitiesmanagementapi.client.prisonersearchapi.model.Prisoner
+import uk.gov.justice.digital.hmpps.hmppsactivitiesmanagementapi.model.response.EarliestReleaseDate
+
+fun determineEarliestReleaseDate(prisoner: Prisoner): EarliestReleaseDate {
+  val releaseDate = when {
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/util/EarliestReleaseDateHelper.kt:7]: Is is me or is this repeated in the constructor in the return statement?

**Pull request merged**